test/utils/docker/centos7/Dockerfile

```diff
@@ -17,15 +17,20 @@ RUN yum clean all && \
     bzip2 \
     dbus-python \
     file \
+    gcc \
     git \
     iproute \
+    libffi \
+    libffi-devel \
     make \
     mariadb-server \
     mercurial \
     MySQL-python \
     openssh-clients \
     openssh-server \
+    openssl-devel \
     python-coverage \
+    python-devel \
     python-httplib2 \
     python-jinja2 \
     python-keyczar \
```

Comment:
Do we want to install both pycrypto and cryptography and run the unittests with each library?  (When we code the alternatives, we should be able to do that via a global HAS_CRYPTOGRAPHY/HAS_PYCRYPTO switch in the module and then have the unittest monkeypatch those defines.